packages/react-native/native-js-interface.md

```diff
@@ -4,27 +4,143 @@ This is a working spec of this the interface that the native layer (`BugsnagReac
 
 ## Intro
 
-`BugsnagReactNative` is a class which will be instantiated once for the entire runtime and accessible via JS. It's responsible for managing the instantiation of the native client (either `bugsnag-android` or `bugsnag-cocoa`) and providing forwarding on data and invocations from JS to the native client.
+`BugsnagReactNative` is a class which will be instantiated once for the entire runtime and accessible via JS. It's responsible for managing the communicating with the native client (either `bugsnag-android` or `bugsnag-cocoa`) and providing data and invocations from JS to the native client.
 
 ## Accessing `BugsnagReactNative` from JavaScript
 
 Access is set up in the standard React Native way<sup>[1](https://facebook.github.io/react-native/docs/0.59/native-modules-ios),[2](https://facebook.github.io/react-native/docs/0.59/native-modules-android)</sup> such that `BugsnagReactNative` is available in JavaScript like so:
 
 ```js
 const { NativeModules } = require('react-native')
-const NativeClient = NativeModules.BugsnagReactNative
+const { BugsnagReactNative } = NativeModules
 ```
 
 ## API
 
-### `BugsnagReactNative.leaveBreadcrumb(breadcrumb): void`
+### Methods
+
+#### `BugsnagReactNative.getConfig(): configMap`
+
+Retrieves a data structure representing the configuration which the native client was initialised with. The data structure needn't be immutable since it will be passed over the bridge anyway, but mutations should not have any effect.
+
+An "empty" map will indicate to the JS layer that the native client is not configured, which is a problem.
+
+This method is required to be synchronous so it should be annotated as such. This is because during its synchronous configuration, the JS layer needs to obtain config from the native layer.
```

Comment:
I'm concerned about the ramifications of making this method call synchronous. The only reason we need it to be synchronous is that the interface of the JS notifier is synchronous. It would be a big change and departure from the current interface to make it async, but is that a good enough reason?

Reply:
Seems reasonable to make this synchronous as it's required for init.